logstash-core/lib/logstash/filter_delegator.rb

```diff
@@ -37,7 +37,9 @@ def config_name
     def multi_filter(events)
       @metric_events.increment(:in, events.size)
 
+      start = Time.now
       new_events = @filter.multi_filter(events)
+      @filter.metric.increment(:time_ms, ((Time.now - start)*1000).to_i)
```

Comment:
@jordansissel If you don't provide a proc/lambda you already get a stop watch class.
See https://github.com/elastic/logstash/blob/master/logstash-core/spec/logstash/instrument/metric_spec.rb#L85